store/cachekv/store.go

```diff
@@ -178,23 +182,53 @@ func (store *Store) iterator(start, end []byte, ascending bool) types.Iterator {
 	}
 
 	store.dirtyItems(start, end)
-	cache = newMemIterator(start, end, store.sortedCache, ascending)
+	cache = newMemIterator(start, end, store.sortedCache, store.deleted, ascending)
 
 	return newCacheMergeIterator(parent, cache, ascending)
 }
 
+// strToByte is meant to make a zero allocation conversion
+// from string -> []byte to speed up operations, it is not meant
+// to be used generally, but for a specific pattern to check for available
+// keys within a domain.
+func strToByte(s string) []byte {
```

Comment:
I believe this is not new code, and actually was written before? (I don't think that either I or joon wrote it) 

Maybe it just got deleted at some point?

Reply:
Please check internal/conv, we have an exported function for this code.

Reply:
ah perfect, thanks